### PR TITLE
added Rotate List problem for CTCI LL session

### LINKED ISSUE #11: Linked List: Write-up for question 3

### PR DESCRIPTION
**Pull request by etcadinfinitum**
Solution is implemented in C++. Could use some more tests?

Fixes #11

### CODE REVIEW

File: Spring-2019/linked_lists/SOLUTIONS_2019Spring.md [hunk 1]
```diff
@@ -39,21 +39,35 @@ If the problem is simple enough, remove this section. TODO :bug:
...
-### 3. PROBLEM 3 TODO :bug:
+### 3. Rotate List 
 
-Source: TODO :bug:
+Source: [CrazyForCode](http://www.crazyforcode.com/rotate-linked-list-k-nodes/)
 
 #### Scenario
 
-Problem Statement TODO :bug:
+Given a singly linked list, rotate the list by `k` nodes.
```
Comment (bpas247) [inline Spring-2019/linked_lists/SOLUTIONS_2019Spring.md]: Is it rotating it clock-wise? counter clock-wise?

File: Spring-2019/linked_lists/SOLUTIONS_2019Spring.md [hunk 2]
```diff
@@ -39,21 +39,35 @@ If the problem is simple enough, remove this section. TODO :bug:
...
+Java:
+
+```java
+public ListNode rotate(ListNode node, int k) { }
+```
+
+C++:
+
+```c++
+Node<T>* rotate(Node<T>* node, int k);
```
Comment (bpas247) [inline Spring-2019/linked_lists/SOLUTIONS_2019Spring.md]: You should also include the class definitions for `Node<T>` and `ListNode`

File: Spring-2019/linked_lists/SOLUTIONS_2019Spring.md [hunk 3]
```diff
@@ -89,20 +103,67 @@ TODO :bug:
...
-TODO :bug:
-
-#### Testing The Solutions OR Driver For Solution
-
-TODO :bug:
-
-
+_Note: An important question to ask with this question is whether `k` is 
+guaranteed to be smaller than the length of the list. This implementation 
+assumes this is true._
```
Comment (bpas247) [inline Spring-2019/linked_lists/SOLUTIONS_2019Spring.md]: We should include this as an assumption in the docs problem

File: Spring-2019/linked_lists/rotate_list/rotate_list.cpp [hunk 4]
```diff
@@ -0,0 +1,73 @@
+#include <iostream>
+#include <sstream>
+#include <string>
+
+using namespace std;
+
+template <typename T>
```
Comment (bpas247) [inline Spring-2019/linked_lists/rotate_list/rotate_list.cpp]: I feel like generics makes the solution a bit more convoluted than it needs to be. Maybe just assume that it's a linked list of ints?

Comment (etcadinfinitum) [reply Spring-2019/linked_lists/rotate_list/rotate_list.cpp]: fair enough~

File: Spring-2019/linked_lists/SOLUTIONS_2019Spring.md (hunk 1 above)
Comment (etcadinfinitum) [reply Spring-2019/linked_lists/SOLUTIONS_2019Spring.md]: fixed in 775a59533001aa2787e743085c5b3c7c1ae69c34

File: Spring-2019/linked_lists/SOLUTIONS_2019Spring.md (hunk 3 above)
Comment (etcadinfinitum) [reply Spring-2019/linked_lists/SOLUTIONS_2019Spring.md]: fixed in 775a59533001aa2787e743085c5b3c7c1ae69c34

File: Spring-2019/linked_lists/rotate_list/rotate_list.cpp (hunk 4 above)
Comment (etcadinfinitum) [reply Spring-2019/linked_lists/rotate_list/rotate_list.cpp]: Fixed solution method and driver files in cb831b7e3c703adf1b6403eaa6002e61f9722caa

File: Spring-2019/linked_lists/SOLUTIONS_2019Spring.md (hunk 2 above)
Comment (etcadinfinitum) [reply Spring-2019/linked_lists/SOLUTIONS_2019Spring.md]: fixed in d17460e197dbaec79819eb1d8f0fb19fb1e85b17

**Pull request merged**